apps/server/src/modules/task/controller/task-copy.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Body, Controller, Param, Post } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import { ParseObjectIdPipe } from '@shared/controller/pipe';
+import { ICurrentUser } from '@shared/domain';
+import { Authenticate, CurrentUser } from '@src/modules/authentication/decorator/auth.decorator';
+import { TaskCopyMapper } from '../mapper/task-copy.mapper';
+import { TaskCopyUC } from '../uc/task-copy.uc';
+import { TaskCopyApiParams } from './dto/task-copy.params';
+import { TaskCopyApiResponse } from './dto/task-copy.response';
+
+@ApiTags('TaskCopy')
+@Authenticate('jwt')
+@Controller('tasks/copy')
```

Comment:
to follow the restful principles, I think it would be better to use 'tasks/:id/copy' as route. Also note that we need an id somewhere, and only get one if we define it somewhere in the route (and according to rest we should use 'tasks/:id/copy' over 'tasks/copy/:id'.

Reply:
please note that the first part of that route is defined by the controller decorater, and the rest of the route by the Post decorator.